app/base/static/assets/js/vis/simple-bar-chart.js

```diff
@@ -42,55 +43,111 @@ class SimpleBarChart {
             .attr("dx", "-.8em")
             .attr("dy", ".15em")
             .attr("transform", "rotate(-65)");
 
-        g.append("g")
-            .attr("class", "axis axis--y")
-            .call(d3.axisLeft(y))
+
+        //declare yAxis and yAxis element variable to be used in resize function
+        let yAxis = d3.axisLeft().scale(y);
+        let yAxisEL = g.append("g")
+            .call(yAxis);
+        yAxisEL.attr("class", "axis axis--y")
             .append("text")
             .attr("class", "axis-title");
 
-        g.selectAll(".bar")
+        //declare bar variable to be used in resize function
+        let bar = g.selectAll(".bar")
             .data(data)
-            .enter().append("rect")
-            .attr("x", function (d) {
+            .enter().append("rect");
+
+        bar.attr("x", function (d) {
+            return x(d.index);
+        })
+        .attr("y", function (d) {
+            return y(d[field]);
+        })
+        .attr("width", x.bandwidth())
+        .attr("height", function (d) {
+            return height - y(d[field]);
+        })
+        .attr("fill", '#4682B4')
+        .on("mousemove", function(d){
+            tooltip_barchart
+                .style("left", d3.event.pageX - 50 + "px")
+                .style("top", d3.event.pageY - 70 + "px")
+                .style("display", "inline-block")
+                .html(d.index + '-> ' + d[field]);
+        })
+        .on("mouseout", function(d){ tooltip_barchart.style("display", "none");});
+
+        let gap = this.GAP;
+
+        //declare resize function
+        function resize() {
+            let h = window.innerHeight - margin.top - margin.bottom;
+            let card = document.getElementById('charts');
+            let w = card.offsetWidth - gap - gap;
+
+            canvas.style.width = card.offsetWidth + "px";
+            canvas.style.height = (h + 20) + "px";
+
+            //resize svg size
+            svg.attr("width", card.offsetWidth)
+                .attr("height", h);            
+
+            //resize rect
+            let rectEL = document.getElementById('rect');
+            rectEL.setAttribute("width", w);
+            rectEL.setAttribute("height", h);
+
+            //update x and y range
+            x.range([0, w - 100]);
+            y.range([h - 100, 0]);
+
+            //rescale
+            xAxis.scale(x);
+            yAxis.scale(y);
+
+            //update axis element
+            xAxisEL.attr("transform", "translate(0," + (h - 100) + ")")
+                .call(xAxis);
+            yAxisEL.call(yAxis);
+
+            //update data
+            bar.attr("x", function (d) {
                 return x(d.index);
             })
             .attr("y", function (d) {
                 return y(d[field]);
             })
             .attr("width", x.bandwidth())
             .attr("height", function (d) {
-                return height - y(d[field]);
+                return h - 100 - y(d[field]);
```

Comment:
magic number 100?

Reply:
get rid of this number 100, instead of return h - margin.top -margin.bottom - y(d[field]);